book/src/ui/introduction.md

```diff
@@ -0,0 +1,42 @@
+# Introduction 
+
+The most usual way to have your players interact with your game is through your user interface or UI.
+In general UI includes all kinds of widgets from images, buttons, progress bars, text, sliders, popup menus, etc.
+
+The API in Amethyst was designed more to provide users with building blocks for the UI and without 
+a specific layout system. The reason is that you don't often see, if any, layout systems 
+used in games, although are very popular GUI frameworks. 
+
+Please note that not all forementinoted widgets exist in Amethyst yet.
+
+## Setting up the UI
+
+The first thing you need to add to your systems is the [UiBundle](https://docs.amethyst.rs/master/amethyst_ui/struct.UiBundle.html). The `UiBundle` registers
+all the needed components,systems and resources in order to be able use the UI. Another **very important thing**
+is that you want to add the [InputBundle](https://docs.amethyst.rs/master/amethyst_input/struct.InputBundle.html) 
+**before** the `UiBundle`,
+otherwise the application will panic, since it is a dependency for the `UiBundle`!
+
+Now you are able to create your widgets! Unfortunately you won't be able to see them. That's why you also need 
+to add a plugin to your rendering bundle called [RenderUi](https://docs.amethyst.rs/master/amethyst_ui/struct.RenderUi.html) in order
+to draw these widgets.
+
+A minimalistic game data would now look like this:  
+
+```rust
+let game_data = GameDataBuilder::default()
+        .with_bundle(InputBundle::<StringBindings>::new())?
+        .with_bundle(UiBundle::<StringBindings>::new())?
+        .with_bundle(
+            RenderingBundle::<DefaultBackend>::new()
+                .with_plugin(
+                    RenderToWindow::from_config_path(display_config_path)?
+                        .with_clear([0.0, 0.0, 0.0, 1.0]),
+                )
+                .with_plugin(RenderUi::default()),
+        )?;
```

Comment:
CI is saying you need to import the stuff. Seems like what you need is `,edition2018,no_run,noplaypen` after the `rust` at the top so that the CI doesn't try to run your code.

Reply:
Oof, sorry. The `rust` at the top part is like the MD's way of starting code: \`\`\`rust\`\`\`

Reply:
Thanks, fixed it!

Reply:
@ab0v3g4me Sure, thanks for writing this! Plus, seems like bors is the only one that can make the current bors pass. What you've done at least let the CI pass.